rfcs/0000-dns-cluster-aware-delegation.md

```diff
@@ -0,0 +1,160 @@
+# DNS Cluster Aware Delegation
+
+- Feature Name: `dns_cluster_aware_delegation`
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Proposal to add functionality to delegate the processing of a DNSRecord to a designated cluster or clusters in a multi cluster environment. 
+
+# Motivation
+[motivation]: #motivation
+
+Multi cluster DNS is currently achieved by using the eventual provider DNS service (AWS Route etc ..) as a store for ownership metadata using specially created TXT records, and as a central API service that all clusters can communicate with.
+Each cluster processes its own DNSRecords, becoming aware of other DNSRecords contributing to the same set of endpoints via this centrally stored data, in turn allowing it to correctly translate the DNSRecord endpoints into an appropriate API operation (Create/Update/Delete) and get to the desired state. 
+
+In some cases, such as our current CoreDNS solution, there is no central off cluster API service that can be configured on all clusters and as such a different approach to multiple cluster discovery and reconciliation is required.
+
+* Add a provider agnostic alternative for multi cluster dns support while still maintaining current behaviour for existing providers.
+* Reduce the number of required CoreDNS instances to one, but still allow multiple if required, redundancy etc...
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Delegating Authority to Publish
+
+Delegating authority is a concept whereby you can optionally specify that you do not want each DNSPolicy to publish its endpoints directly to a provider, but instead have a central authoritative DNSRecord be created that will take care of publishing all endpoints for a particular set of records.
+
+This feature is controlled by the `authorityDelegation` field in the DNSPolicy spec, which has a single required field `role` which must be one of `primary` or `remote`.
+
+- `spec.authorityDelegation` when set signals that records produced by this policy will have their endpoints published via an authoritative record
+- `spec.authorityDelegation.role` signals what role this particular policy is taking, it is expected there should be at least one `primary`
```

Comment:
is there any way for the controllers to determine that no primary exists and set a status on the policy?

Reply:
Yeah, well i think at a minimum we would want that and i did mention something [here](https://github.com/Kuadrant/architecture/pull/127/files#diff-4c24f82bc117f776d24141a6246aff55e71aaa9d5bf560c30dda501c3ca625fbR44) which is effectively the same, no primary no authoritative